File: src/main/java/com/braintreegateway/DescriptorRequest.java
package com.braintreegateway;

public abstract class DescriptorRequest extends BaseRequest {

    protected String name;
    protected String phone;

    public DescriptorRequest() {
        super();
    }
    
    public DescriptorRequest name(String name) {
        this.name = name;
        return this;
    }

    public DescriptorRequest phone(String phone) {
        this.phone = phone;
        return this;
    }

    @Override
    public String toXmlString() {
        return buildRequest("descriptor").toXmlString();
    }

    @Override
    public String toQueryString(String root) {
        return buildRequest(root).toQueryString();
    }

    protected RequestBuilder buildRequest(String root) {
        return new RequestBuilder(root).
            addElement("name", name).
            addElement("phone", phone);
    }
}

File: src/main/scala/com/braintreegateway/DescriptorRequest.scala
package com.braintreegateway

class DescriptorRequest[P <: Request](val done:P) extends BaseRequest with HasParent[P] {

  protected var name: Option[String] = None
  protected var phone: Option[String] = None

  def name(name: String): this.type = {
    this.name = Option(name)
    this
  }

  def phone(phone: String): this.type = {
    this.phone = Option(phone)
    this
  }

  override def toXmlString: String = {
    buildRequest("descriptor").toXmlString
  }

  override def toQueryString(root: String): String = {
    buildRequest(root).toQueryString
  }

  protected def buildRequest(root: String): RequestBuilder = {
    new RequestBuilder(root).addElement("name", name).addElement("phone", phone)
  }
}

object DescriptorRequest {
  def apply[P <: Request](parent: P) = new DescriptorRequest[P](parent)
}